### PR TITLE
[wasm] Improve the autogenerated project

### PR DESCRIPTION
**Pull request by radical**
1. Currently, we autogenerate a wrapper wasm project for the actual benchmark project. And the actual project has no good way of controlling this generated project, which means that we end up patching the template project in this repo.

To fix this:

- Automatically import any `$projectName.Wasm.props`, and
  `$projectName.Wasm.targets` files, if found.
  - This allows the actual project to control settings for the generated
    project, eg, for modifying the wasm build to add some files to the
    wasm vfs.

2. Remove `runtimeSrcDir` command line parameter, to enable switching to using wasm workloads which contain all the relevant bits.

### CODE REVIEW

**Comment by radical** [comment]
cc @loopedbard3 @naricc @radekdoulik @lewing @drewscoggins

File: src/BenchmarkDotNet/ConsoleArguments/ConfigParser.cs [hunk 1]
```diff
@@ -131,12 +131,6 @@ private static bool Validate(CommandLineOptions options, ILogger logger)
                 {
                      logger.WriteLineError($"The provided {nameof(options.AOTCompilerPath)} \"{ options.AOTCompilerPath }\" does NOT exist. It MUST be provided.");
                 }
-                else if (MonikerIsWasm(runtimeMoniker) && (options.RuntimeSrcDir == null || options.RuntimeSrcDir.IsNotNullButDoesNotExist()))
```
Comment (radekdoulik) [inline src/BenchmarkDotNet/ConsoleArguments/ConfigParser.cs:134]: You can remove the `MonikerIsWasm` method too, as it was only used here IIRC.

**Review by radekdoulik — APPROVED** [review]
Besides the `MonikerIsWasm` method it LGTM.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @radical !

**Comment by radical** [comment]
The [AppVeyor](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/builds/42855904) seems to have failed:

```
dotnet.exe : [xUnit.net 00:18:52.90]     AwaitingTasksShouldNotInterfereAllocationResults(toolchain: InProcessEmitToolchain) [FAIL]
[14913](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14913)At C:\projects\benchmarkdotnet\build.ps1:89 char:1
[14914](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14914)+ & dotnet run --project build/Build.csproj -- $args
[14915](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14915)+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[14916](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14916)    + CategoryInfo          : NotSpecified: ([xUnit.net 00:1...olchain) [FAIL]:String) [], RemoteException
[14917](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14917)    + FullyQualifiedErrorId : NativeCommandError
[14918](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14918) 
[14919](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14919)  Failed AwaitingTasksShouldNotInterfereAllocationResults(toolchain: InProcessEmitToolchain) [11 s]
[14920](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14920)  Error Message:
[14921](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14921)   Assert.Equal() Failure
[14922](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14922)Expected: 0
[14923](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14923)Actual:   100864
[14924](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14924)  Stack Trace:
[14925](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14925)     at BenchmarkDotNet.IntegrationTests.MemoryDiagnoserTests.AssertAllocations(IToolchain toolchain, Type benchmarkType, Dictionary`2 benchmarksAllocationsValidators) in C:\projects\benchmarkdotnet\tests\BenchmarkDotNet.IntegrationTests\MemoryDiagnoserTests.cs:line 261
[14926](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/history#L14926)   at BenchmarkDotNet.IntegrationTests.MemoryDiagnoserTests.AwaitingTasksShouldNotInterfereAllocationResults(IToolchain toolchain) in C:\projects\benchmarkdotnet\tests\BenchmarkDotNet.IntegrationTests\MemoryDiagnoserTests.cs:line 150
```

@adamsitnik  Is this the reason for no new nuget being generated? I tried `nuget list -AllVersions BenchmarkDotNet`, and I don't see anything newer than `0.13.1.1706`. Please correct me if I'm not looking at this correctly.

**Comment by adamsitnik** [comment]
@radical you are right, the CI has not produced the package due to this test failure (it's a noisy test, not an issue in the product itself)

**Pull request merged**